bin/configure_ssh.sh

```diff
@@ -0,0 +1,40 @@
+#!/bin/bash
+
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+# Makes a .ssh dir if one does not exist and adds a $HOME/.ssh/config file.
+# The secret fetched from Taskcluster is a private SSH key. It will be
+# stored in $HOME/.ssh/id_rsa_experiments_pusher (see ./bin/get_secret.js)
+# and that file is pointed at by the $HOME/.ssh/config file. Once we have
+# this key and SSH config set up, we can push to the Try server!
+
+set -e
+
+echo ">>> Make the $HOME/.ssh dir if it doesn't exist"
+cd ~
+if [ ! -d .ssh ]
+then
+  mkdir .ssh
+fi
+
+echo ">>> Set up SSH config with Experiments Pusher user"
+if [ ! -f ~/.ssh/config ]
+then
+  cat <<EOF > ~/.ssh/config
+Host hg.mozilla.org
+  User experiments-pusher@mozilla.com
+  # for experiments try server pushing
+  IdentityFile ~/.ssh/id_rsa_experiments_pusher
+EOF
+fi
+
+# Without this, the first time I connect to this server, it will prompt me
+# to ask if I am sure I want to connect, and the task will hang and timeout.
```

Comment:
I'm not an expert, but as I understand it this is because someone might be impersonating hg.mozilla.org, and what you want here is not to just accept whatever hg tells you when this script runs, but instead to independently get the right hash for the hg.mozilla.org server, and then encode that hash as part of these build scripts. That way, you won't accidentally connect and push to a fraudulent hg.mozilla.org server.

Having said that, I have no idea how to do any of that specifically. I only really understand the concept.

Reply:
You're right; I didn't think about that. I talked to Dan Veditz (Security Engineering) a bit about this:

In the SSH security model, it's a reasonable thing to SSH to a host server one time. Once this first request is made, you keep the host's public key and compare it for every subsequent request. That way, if the key changes, you will know to be suspicious.

Right now I am just getting this key every time and assuming it's trustworthy. I could be sending my credentials to a spoofed server. This is what is known as a "man-in-the-middle" attack, where in this case they could change what I am committing to the Try server while the connection is open.

What I can do instead is hard code hg.mozilla.org's public key and compare what I get when making the request to that static value each time. Importantly, I need to get this public key "out of band" from a trustworthy source first. I probably have it stored in my local ssh files, as I have pushed to the Try server previously from my local machine with my own LDAP account.

In terms of getting this PR a security review, Dan suggested that I contact Julien Vehent (:ulfh).